src/main/java/org/llorllale/cactoos/matchers/HasValuesMatching.java

```diff
@@ -48,42 +44,24 @@
  *
  * @param <X> Type of item.
  * @since 1.0.0
- * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
  */
-public final class HasValuesMatching<X> extends
-    TypeSafeDiagnosingMatcher<Iterable<X>> {
-
-    /**
-     * The function to match at least one element within the {@link Iterable}.
-     */
-    private final Func<X, Boolean> fnc;
-
+public final class HasValuesMatching<X> extends MatcherEnvelope<Iterable<X>> {
     /**
      * Ctor.
      * @param fnc The function to match at least one element within the
      *  {@link Iterable}.
      */
     public HasValuesMatching(final Func<X, Boolean> fnc) {
-        super();
-        this.fnc = fnc;
-    }
-
-    @Override
-    public void describeTo(final Description dsc) {
-        dsc.appendText("The function matches at least 1 element.");
-    }
-
-    @Override
-    protected boolean matchesSafely(final Iterable<X> actual,
-        final Description dsc) {
-        dsc.appendText(
-            new UncheckedText(
-                new FormattedText(
-                    "No any elements from [%s] matches by the function",
-                    new TextOf(actual)
+        super(
+        // @checkstyle IndentationCheck (7 line)
+        actual -> new Or(fnc, actual).value(),
+        desc -> desc.appendText("The function matches at least 1 element."),
+        (actual, desc) -> desc.appendText(
+                    new FormattedText(
+                        "No any elements from [%s] matches by the function",
+                        new TextOf(actual)
+                    ).asString()
```

Comment:
@victornoel is there a particular reason not to respect standard indentation rules here?

Reply:
@scristalli see https://github.com/llorllale/cactoos-matchers/pull/134#discussion_r303199484, but I will add one level of indentation